shell/platform/embedder/embedder_external_view_embedder.cc

```diff
@@ -122,13 +121,309 @@ static FlutterBackingStoreConfig MakeBackingStoreConfig(
+                                                                region);
+  }
+
+  /// Returns the deepest layer to which the platform view can be added. That
+  /// would be (whichever comes first):
+  /// - First layer from back that has platform view that intersects with this
+  ///   view
+  /// - Very last layer from back that has surface that doesn't intersect with
+  ///   this. That is because layer content renders on top of the platform view.
+  Layer& GetLayerForPlatformView(PlatformView view) {
+    for (auto iter = layers_.rbegin(); iter != layers_.rend(); ++iter) {
+      // This layer has surface that intersects with this view. That means we
+      // went one too far and need the layer before this.
+      if (iter->IntersectsFlutterContents(view.clipped_frame)) {
+        if (iter == layers_.rbegin()) {
+          layers_.emplace_back();
+          return layers_.back();
+        } else {
+          --iter;
+          return *iter;
+        }
+      }
+      if (iter->IntersectsPlatformView(view.clipped_frame)) {
+        return *iter;
+      }
+    }
+    return layers_.front();
+  }
+
+  /// Finds layer to which the Flutter content can be added. That would
+  /// be first layer from back that has any intersection with this region.
+  Layer& GetLayerForFlutterContentsRegion(const DlRegion& region) {
+    for (auto iter = layers_.rbegin(); iter != layers_.rend(); ++iter) {
+      if (iter->IntersectsPlatformView(region) ||
+          iter->IntersectsFlutterContents(region)) {
+        return *iter;
+      }
+    }
+    return layers_.front();
+  }
+
+  std::vector<Layer> layers_;
+  SkISize frame_size_;
+};
+
+};  // namespace
+
 void EmbedderExternalViewEmbedder::SubmitFrame(
     GrDirectContext* context,
     const std::shared_ptr<impeller::AiksContext>& aiks_context,
     std::unique_ptr<SurfaceFrame> frame) {
-  auto [matched_render_targets, pending_keys] =
-      render_target_cache_.GetExistingTargetsInCache(pending_views_);
+  SkRect _rect = SkRect::MakeIWH(pending_frame_size_.width(),
+                                 pending_frame_size_.height());
+  pending_surface_transformation_.mapRect(&_rect);
+
+  LayerBuilder builder(SkISize::Make(_rect.width(), _rect.height()));
+
+  for (auto view_id : composition_order_) {
+    auto& view = pending_views_[view_id];
+    builder.AddExternalView(view.get());
+  }
+
+  builder.PrepareBackingStore([&](FlutterBackingStoreConfig config) {
+    std::unique_ptr<EmbedderRenderTarget> target;
+    if (!avoid_backing_store_cache_) {
+      target = render_target_cache_.GetRenderTarget(
+          EmbedderExternalView::RenderTargetDescriptor(
+              SkISize{static_cast<int32_t>(config.size.width),
+                      static_cast<int32_t>(config.size.height)}));
+    }
+    if (target != nullptr) {
+      return target;
+    } else {
+      return create_render_target_callback_(context, aiks_context, config);
+    }
```

Comment:
Similar comment to before; we can just leave the if as a bailout and unwrap the else to the default path for the function body after the if.